cli/src/commands/move.rs

```diff
@@ -89,6 +89,13 @@ pub(crate) fn cmd_move(
         source.id().hex(),
         destination.id().hex()
     );
+
+    // If the user provided --interactive or files, the intent of the user was
+    // clearly to work on files rather than on commits, - it would be strange
+    // if the user wrote jj squash foo.rs and found that the branch pointer had
+    // moved to the parent, for example.
+    let abandon = !matches!(diff_selector, DiffSelector::Interactive(_)) && args.paths.is_empty();
```

Comment:
I don't agree that the user clearly intended to leave the source commit behind. Maybe they just wanted to review the changes in their diff editor before confirming? I think it's a little unfortunate to lose that property, but maybe it's fine. I would prefer to not merge this just before the release later today.

FWIW, I view squash with paths or `-i` as a way of doing a partial squash; I don't think of it as "working on files" vs "working on  commits".

Reply:
Changed to no longer infer behavior from flags, so I'll mark this a resolved.